pkg/machines/components/vm/disks/diskAdd.jsx

```diff
@@ -379,6 +411,26 @@ export class AddDiskModalBody extends React.Component {
             this.setState({ busType: value });
             break;
         }
+        case 'file': {
+            stateDelta.file = value;
+            this.setState(prevState => {
+                if (value.endsWith(".iso"))
+                    this.onValueChanged("device", "cdrom");
+                return stateDelta;
+            });
+            break;
+        }
+        case 'device': {
+            stateDelta.device = value;
+            this.setState(prevState => {
+                // Disk device "cdrom" and bus "virtio" are uncompatible, see:
+                // https://www.redhat.com/archives/libvir-list/2019-January/msg01104.html
+                if (value === "cdrom" && prevState.busType === "virtio")
+                    this.onValueChanged("busType", "scsi");
```

Comment:
virt-manager chooses SATA by default. What's the background between this deviation?

Reply:
What virtmanager does is look if a a disk with same device type exists. If so, virtmanager uses its bus (that approach I can use too). If not, it just let's the first bus form the alphabetically ordered list of compatible buses. (that I din't think I should do)

source: https://github.com/virt-manager/virt-manager/blob/master/virtManager/addhardware.py#L1056

Reply:
Also I found this: "If omitted, the bus type is inferred from the style of the device name (e.g. a device named 'sda' will typically be exported using a SCSI bus)". It makes me feel like scsi bus should be default option since libvirt defaults to it too.
source: https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms

So my idea is to combine what virtmanager does and what lib doc suggests:
- Look if a disk with same device already exists on the VM. If so, use the same bus.
- If no such disk exists, and use is switching from cdrom with bustype virtio to another device type where virtio bus type is not present, then default to scsi. (as per libvirt dov)
- else If user is switching between device types, where already selected bus types is present for both device types, then keep that bus type.

Reply:
Sounds good. You can also just allow the user to select from the available bus types and default to the alphabetically first choice. It's up to you.
Feel free to to that as a follow up as well, how it is now is not wrong.